src/renderers/wgpu/mod.rs

```diff
@@ -0,0 +1,552 @@
+mod glow {}
+
+use std::sync::Arc;
+use std::{any::TypeId, cell::RefCell, collections::HashMap};
+
+use wgpu::{Device, Queue, RenderPipeline, Surface, Texture};
+
+use crate::{
+    model::ModelTexture,
+    nodes::{
+        node::{ExtInoxNode, InoxNodeUuid},
+        node_data::{BlendMode, Composite, InoxData, Mask, Part},
+        node_tree::ExtInoxNodeTree,
+    },
+};
+
+use self::{texture::load_texture, vbo::Vbo};
+
+pub mod app;
+pub mod shader;
+pub mod texture;
+pub mod vbo;
+
+// lib.rs
+#[repr(C)]
+#[derive(Copy, Clone, Debug)]
+pub struct Vertex {
+    position: [f32; 3],
+    color: [f32; 3],
```

Comment:
I think `glam` types would work here, `Vec3` and all.

```suggestion
    position: Vec3,
    color: Vec3,
```